cognite/client/_api/data_modeling/streams.py

```diff
@@ -0,0 +1,180 @@
+        }
+
+    @overload
+    def __call__(
+        self,
+        chunk_size: None = None,
+        limit: int | None = None,
+    ) -> Iterator[Stream]: ...
+
+    @overload
+    def __call__(
+        self,
+        chunk_size: int,
+        limit: int | None = None,
+    ) -> Iterator[StreamList]: ...
+
+    def __call__(
+        self,
+        chunk_size: int | None = None,
+        limit: int | None = None,
+    ) -> Iterator[Stream] | Iterator[StreamList]:
+        """Iterate over streams
+
+        Fetches streams as they are iterated over, so you keep a limited number of streams in memory.
+
+        Args:
+            chunk_size (int | None): Number of streams to return in each chunk. Defaults to yielding one stream a time.
+            limit (int | None): Maximum number of streams to return. Defaults to returning all items.
+
+        Returns:
+            Iterator[Stream] | Iterator[StreamList]: yields Stream one by one if chunk_size is not specified, else StreamList objects.
+        """
+        return self._list_generator(
+            list_cls=StreamList,
+            resource_cls=Stream,
+            method="GET",
+            chunk_size=chunk_size,
+            limit=limit,
+            headers=self.__alpha_headers,
+        )
+
+    def __iter__(self) -> Iterator[Stream]:
+        """Iterate over streams
+
+        Fetches streams as they are iterated over, so you keep a limited number of streams in memory.
+
+        Returns:
+            Iterator[Stream]: yields Streams one by one.
+        """
+        return self()
+
+    def retrieve(self, external_id: str) -> Stream | None:
+        """`Retrieve a stream. <https://developer.cognite.com/api#tag/Streams/operation/byStreamIdsStreams>`_
+
+        Args:
+            external_id (str): No description.
+
+        Returns:
+            Stream | None: Requested stream or None if it does not exist.
+
+        Examples:
+
+                >>> from cognite.client import CogniteClient
+                >>> client = CogniteClient()
+                >>> res = client.data_modeling.streams.retrieve(streams='myStream')
+
+            Get multiple streams by id:
+
+                >>> res = client.data_modeling.streams.retrieve(streams=["MyStream", "MyAwesomeStream", "MyOtherStream"])
+
+        """
+        identifier = Identifier.load(external_id=external_id)
+        return self._retrieve(identifier=identifier, cls=Stream, headers=self.__alpha_headers)
+
+    def delete(self, external_id: str) -> None:
+        """`Delete one or more streams <https://developer.cognite.com/api#tag/Streams/operation/deleteStreamsV3>`_
+
+        Args:
+            external_id (str): ID of streams.
+        Examples:
+
+            Delete streams by id:
+
+                >>> from cognite.client import CogniteClient
+                >>> client = CogniteClient()
+                >>> client.data_modeling.streams.delete(streams=["myStream", "myOtherStream"])
+        """
```

Comment:
note: Streams api deviate from cognite api standard on delete (bug has been opened on this), which is generally to do post with items list under , instead we have a issue a DELETE request.

Reply:
we only accept one item in the list of streams to be deleted and created btw..